gneiss/util.py

```diff
@@ -184,3 +188,81 @@ def rename_internal_nodes(tree, names=None, inplace=False):
             n.name = label
             i += 1
     return _tree
+
+
+def _intersect_of_table_metadata_tree(table, metadata, tree):
+    """ Matches tips, features and samples between the table, metadata
+    and tree.  This module returns the features and samples that are
+    contained in all 3 objects.
+
+    Parameters
+    ----------
+    table : pd.DataFrame
+        Contingency table where samples correspond to rows and
+        features correspond to columns.
+    metadata: pd.DataFrame
+        Metadata table that contains information about the samples contained
+        in the `table` object.  Samples correspond to rows and covariates
+        correspond to columns.
+    tree : skbio.TreeNode
+        Tree object where the leaves correspond to the columns contained in
+        the table.
+
+    Returns
+    -------
+    pd.DataFrame
+        Subset of `table` with common row names as `metadata`
+        and common columns as `tree.tips()`
+    pd.DataFrame
+        Subset of `metadata` with common row names as `table`
+    skbio.TreeNode
+        Subtree of `tree` with common tips as `table`
+    """
+    if np.any(table <= 0):
+        raise ValueError('Cannot handle zeros or negative values in `table`. '
+                         'Use pseudocounts or ``multiplicative_replacement``.'
+                         )
+    # check to see if there are overlapping nodes in tree and table
+    overlap = {n.name for n in tree.traverse()} & set(table.columns)
```

Comment:
Yup - the only problem is we specifically need strings for the names rather than the entire `TreeNode`.  So we'll still need the set comprehension in the end.